301-availability-set-elastic-storage-ecs/azuredeploy.json

```diff
@@ -0,0 +1,726 @@
+        4,
+        5,
+        6,
+        7,
+        8
+      ],
+      "metadata": {
+        "description": "Number of VMs to deploy, limit 5 since this sample is using a single storage account"
+      }
+    },
+    "dataDiskCount": {
+      "type": "int",
+      "defaultValue": 3,
+      "allowedValues": [
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        8
+      ],
+      "metadata": {
+        "description": "NUmber of ECS Data Disks"
+      }
+    },
+    "dataDiskSize": {
+      "type": "int",
+      "defaultValue": 150,
+      "minValue": 150,
+      "maxValue": 1023,
+      "metadata": {
+        "description": "Size of the Data Disk in GB"
+      }
+    },
+    "vmNamePrefix": {
+      "type": "string",
+      "defaultValue": "ecs",
+      "metadata": {
+        "description": "VM name prefix"
+      }
+    },
+    "vmSize": {
+      "allowedValues": [
+        "Standard_A4",
+        "Standard_A5",
+        "Standard_D4",
+        "Standard_D5",
+        "Standard_A4M_V2",
+        "Standard_B4MS",
+        "Standard_D12_v2_Promo",
+        "Standard_D13_v2_Promo",
+        "Standard_D14_v2_Promo",
+        "Standard_D4S_V3"
+      ],
+      "defaultValue": "Standard_A4M_V2",
+      "metadata": {
+        "description": "This is the size of your ECS Nodes. Minimum is A/D4"
+      },
+      "type": "string"
+    },
+    "dnsPrefix": {
+      "type": "string",
+      "defaultValue": "[concat('ecs', uniquestring(resourceGroup().id))]",
+      "metadata": {
+        "description": "dns name prefix. this is the external dns prefix to reach your ECS via the loadbalancer"
+      }
+    },
+    "osImagePublisher": {
+      "type": "string",
+      "defaultValue": "OpenLogic",
+      "allowedValues": [
+        "Centos",
+        "OpenLogic"
+      ],
+      "metadata": {
+        "description": "The Image Publisher, use Openlogic for default Marketplace"
+      }
+    },
+    "osImageOffer": {
+      "type": "string",
+      "defaultValue": "CentOS",
+      "allowedValues": [
+        "Centos-7",
+        "CentOS"
+      ],
+      "metadata": {
+        "description": "The Image Offer, use centos for default Marketplace"
+      }
+    },
+    "osImageSKU": {
+      "type": "string",
+      "defaultValue": "7.4",
+      "allowedValues": [
+        "Centos-7.4",
+        "7.4"
+      ],
+      "metadata": {
+        "description": "The Linux version for the VM. Use 7.4 for default Marketplace"
+      }
+    }
+  },
+  "variables": {
+    "availabilitySetName": "[toLower(concat('ECSaSet-', resourceGroup().name))]",
+    "storageAccountType": "Standard_LRS",
+    "vmSize": "[parameters('vmSize')]",
+    "dnsPrefix": "[parameters('dnsPrefix')]",
+    "osImageVersion": "latest",
+    "addressPrefix": "10.0.0.0/16",
+    "virtualNetworkName": "[tolower(concat('vNet-',resourceGroup().name))]",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
```

Comment:
remove this

Reply:
just learning. used this example as a reference: azure-quickstart-templates/101-loadbalancer-with-nat-rule/azuredeploy.json
When expecting changes, please make sure examples are consistent